clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -469,6 +471,204 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+        Cluster cluster = TestUtils.singletonCluster(topic, 1);
+        Node node = cluster.nodes().get(0);
+        client.setNode(node);
+        Metadata metadata = new Metadata(0, Long.MAX_VALUE);
+        metadata.update(cluster, time.milliseconds());
+        PartitionAssignor assignor = new RoundRobinAssignor();
+
+        final KafkaConsumer<String, String> consumer = newConsumer(time, client, groupId, metadata, assignor,
+                sessionTimeoutMs, heartbeatIntervalMs, Boolean.TRUE, autoCommitIntervalMs);
+        consumer.subscribe(Arrays.asList(topic), new ConsumerRebalanceListener() {
+            @Override
+            public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
+
+            }
+
+            @Override
+            public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+                // set initial position so we don't need a lookup
+                for (TopicPartition partition : partitions)
+                    consumer.seek(partition, 0);
+            }
+        });
+
+        // lookup coordinator
+        client.prepareResponseFrom(new GroupCoordinatorResponse(Errors.NONE.code(), node).toStruct(), node);
+
+        Node coordinator = new Node(Integer.MAX_VALUE - node.id(), node.host(), node.port());
+
+        // join group
+        client.prepareResponseFrom(joinGroupFollowerResponse(assignor, 1, "memberId", "leaderId", Errors.NONE.code()), coordinator);
+
+        // sync group
+        client.prepareResponseFrom(syncGroupResponse(Arrays.asList(partition), Errors.NONE.code()), coordinator);
+
+        consumer.poll(0);
+
+        // respond to the outstanding fetch so that we have data available on the next poll
+        client.respondFrom(fetchResponse(partition, 0, 5), node);
+        client.poll(0, time.milliseconds());
+
+        // the next poll should return the completed fetch
+        consumer.poll(0);
+
+        // 5 records have been returned to the user yet and auto commit is enabled, so the committed offset should be 5
+        final AtomicBoolean commitReceived = new AtomicBoolean(false);
+        client.prepareResponseFrom(new MockClient.RequestMatcher() {
+            @Override
+            public boolean matches(ClientRequest request) {
+                OffsetCommitRequest commitRequest = new OffsetCommitRequest(request.request().body());
+                OffsetCommitRequest.PartitionData partitionData = commitRequest.offsetData().get(partition);
+                if (partitionData.offset == 5) {
+                    commitReceived.set(true);
+                    return true;
+                }
+                return false;
+            }
+        }, offsetCommitResponse(Collections.singletonMap(partition, Errors.NONE.code())), coordinator);
+
+        consumer.unsubscribe();
+
+        consumer.poll(0);
+
+        assertTrue(commitReceived.get());
+    }
+
+    @Test
+    public void testUnsubscribedPartitionOffsetCommitWithoutAutoCommit() {
```

Comment:
I'm not sure this makes sense. The offsets for each group are isolated, so `consumer2` would actually start from position 0. I think a better test case would be the following:

1. Start a single consumer with autocommit disabled.
2. Read 5 records.
3. Call unsubscribe().
4. Verify that no offset commit request was sent.

To be honest, this might be overkill, but I wouldn't complain if it was present.